Tests/SourceKitLSPTests/SwiftInterfaceTests.swift

```diff
@@ -0,0 +1,154 @@
+import LSPTestSupport
+import SKTestSupport
+import SourceKitLSP
+import XCTest
+
+final class SwiftInterfaceTests: XCTestCase {
+
+  /// Connection and lifetime management for the service.
+  var connection: TestSourceKitServer! = nil
+
+  /// The primary interface to make requests to the SourceKitServer.
+  var sk: TestClient! = nil
+
+  var documentManager: DocumentManager! {
+    connection.server!._documentManager
+  }
+
+  override func setUp() {
+    connection = TestSourceKitServer()
+    sk = connection.client
+    _ = try! sk.sendSync(InitializeRequest(
+      processId: nil,
+      rootPath: nil,
+      rootURI: nil,
+      initializationOptions: nil,
+      capabilities: ClientCapabilities(workspace: nil,
+                                       textDocument: TextDocumentClientCapabilities(
+                                        codeAction: .init(
+                                          codeActionLiteralSupport: .init(
+                                            codeActionKind: .init(valueSet: [.quickFix])
+                                          )),
+                                        publishDiagnostics: .init(codeDescriptionSupport: true)
+                                       )),
+      trace: .off,
+      workspaceFolders: nil))
+  }
+  
+  override func tearDown() {
+    sk = nil
+    connection = nil
+  }
+  
+  func testSystemModuleInterface() throws {
+    let url = URL(fileURLWithPath: "/\(UUID())/a.swift")
+    let uri = DocumentURI(url)
+
+    sk.send(DidOpenTextDocumentNotification(textDocument: TextDocumentItem(
+      uri: uri,
+      language: .swift,
+      version: 1,
+      text: """
+      import Foundation
+      """)))
+    
+    let _resp = try sk.sendSync(DefinitionRequest(
+      textDocument: TextDocumentIdentifier(url),
+      position: Position(line: 0, utf16index: 10)))
+    let resp = try XCTUnwrap(_resp)
+    guard case .locations(let locations) = resp else {
+      XCTFail("Unexpected response: \(resp)")
+      return
+    }
+    XCTAssertEqual(locations.count, 1)
+    let location = try XCTUnwrap(locations.first)
+    XCTAssertTrue(location.uri.pseudoPath.hasSuffix("/Foundation.swiftinterface"))
+    let fileContents = try XCTUnwrap(location.uri.fileURL.flatMap({ try String(contentsOf: $0, encoding: .utf8) }))
+    XCTAssertTrue(fileContents.hasPrefix("import "))
+  }
+  
+  func testOpenInterface() throws {
+    guard let ws = try staticSourceKitSwiftPMWorkspace(name: "SwiftPMPackage") else { return }
+    try ws.buildAndIndex()
+    let call = ws.testLoc("Lib.foo:call")
+    try ws.openDocument(call.url, language: .swift)
+    let workspace = try XCTUnwrap(ws.testServer.server?.queue.sync {
+      try XCTUnwrap(ws.testServer.server?.workspaceForDocument(uri: call.docUri))
+    })
+    let swiftLangServer = try XCTUnwrap(ws.testServer.server?._languageService(for: call.docUri,
+                                                                               .swift,
```

Comment:
This is being rendered really weirdly on GitHub (the `.` is inside the `s` of `swift`). Is this a normal period or some weird Unicode character?

Reply:
Hmm, weird. Must have been a rendering issue in Safari then.